### DISCUSSION TITLE
RabbitMQ Cluster Operator memory resource limit 500Mi

### DISCUSSION THREAD

**Discussion by ilabrovic** [rabbitmq/cluster-operator#1537] [General]
We have deployed the RabbitMQ-cluster-operator 2.6.0 and running into OOMKiller.
It seems to run at 500Mi memory limit hardcoded?
I can t configure the resources of this cluster operator pod since it is managed by the operator.
How can i deal with this?

**Comment by ilabrovic** [comment]
After changing defaults in the Subscription resource, we concluded we need 730Mi memory limit to get the rabbitmq-cluster-operator started. Do you have any clue why we need more than the default 500Mi limit?

**Reply by Zerpet** [reply]
We have seen some reports about abnormal memory consumption, leading to OOM killed cluster operator Pods. We haven't been able to reproduce the issue locally. I have a suspicion/guess that memory consumption may be related to the number of objects the Operator can watch i.e. number of configmaps, secrets, statefulsets, services, etc.

Do you have a rough number about how many configmaps, secrets, statefulsets and services are in your cluster?

**Reply by ilabrovic** [reply]
There are thousands of configmaps and services in our cluster, but they all exist in many/other different namespaces.
Why would RabbitMQ need to know whats running on the cluster?
Only the (rabbitmq) resources in the current namespaces would be enough, would it?

**Reply by mkuratczyk** [reply]
Seems like I can reproduce this by declaring a lot of services (completely unrelated to RMQ). No such issue if there's a lot of ConfigMaps. I will look into this

**Reply by mkuratczyk** [reply]
A change that should fix this has been merged. If you have an option to give it a try, that'd be great. The image is `rabbitmqoperator/cluster-operator:sha-62d64ca`. Note that if you just deploy it to an empty test cluster, you won't see a difference. Assuming the problem is where we think it is, the Kubernetes cluster needs to have quite a lot of objects (earlier I thought the `ConfigMap` wouldn't cause that but they do - mine were just very small).

**Reply by tkoft** [reply]
Is there a way I can determine if this is what's causing the OOM issues in my pods? Looking at the memory usage breakdown, I had thought it was related to quorum queues (which by default take up to 512mb for WAL tables before they're flushed).

**Reply by mkuratczyk** [reply]
This issue is about Operator's memory usage, while you're talking about RabbitMQ's memory usage, so it is a different thing. I will get to [your question](https://github.com/rabbitmq/cluster-operator/discussions/1554) soon

**Reply by ilabrovic** [reply]
Thanks, i will investigate if we have that option to try this image. I'll get back to you if/when we can.
